### ISSUE TITLE
Compile error:  ‘major’ is not a function or function pointer

### ISSUE THREAD

**Issue by Equidamoid** [openSUSE/multipath-tools#30]
Hi, 
Here is the complete error:
```
make -j7 -l8 CC=x86_64-pc-linux-gnu-gcc LIBDM_API_FLUSH=1 SYSTEMD=226
(...)
x86_64-pc-linux-gnu-gcc -c -O2 -g -pipe -Wall -Wextra -Wformat=2 -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fPIC -DLIB_STRING=\"lib64\" -DRUN_DIR=\"run\" -I../libmultipath -I../libmpathpersist -I../libmpathcmd *.c
<command-line>:0:0: warning: "_FORTIFY_SOURCE" redefined
<built-in>: note: this is the location of the previous definition
mpath_persist.c: In function ‘mpath_persistent_reserve_in’:
mpath_persist.c:175:10: error: called object ‘major’ is not a function or function pointer
  major = major(info.st_rdev);
          ^
mpath_persist.c:161:6: note: declared here
  int major, minor;
      ^
mpath_persist.c:176:10: error: called object ‘minor’ is not a function or function pointer
  minor = minor(info.st_rdev);
```
GCC 7.3.0

How is this `int major; major = major(...)` even supposed to work? %)

**Comment by mwilck** [comment]
I'm assuming you're not using glibc? It seems that commit ea436715 failed to fix `mpath_persist.c`.
It shows what you need to do: add `#include <sys/sysmacros.h>`.

This is an upstream bug. I'll submit a patch as soon as I find time.

> How is this int major; major = major(...) even supposed to work?

Under glibc, `major(x)` is a macro with an argument which is translated by the preprocessor to  `gnu_dev_major(x)`. `major`  with no argument isn't touched by cpp.

**Comment by mwilck** [comment]
This is solved by 8b9cda51 and ea436715.
